internal/k8s/entity.go

```diff
@@ -158,6 +158,26 @@ func (e K8sEntity) DeepCopy() K8sEntity {
 	}
 }
 
+// EntitiesWithDependenciesAndRest returns two lists of k8s entities: those that may have dependencies,
+// which we will therefore want to apply first (i.e. namespaces and CRDs -- e.g. trying to create a
+// pod in a nonexistent namespace causes an error); and the rest of the entities.
+func EntitiesWithDependenciesAndRest(entities []K8sEntity) (withDeps, rest []K8sEntity) {
```

Comment:
fwiw, if you have a Pod that depends on a namespace, I would usually interpret "Entity with a dependency" to mean the Pod